price_rule.go

```diff
@@ -0,0 +1,121 @@
+type PriceRuleService interface {
+	Get(int64) (*PriceRule, error)
+	Create(PriceRule) (*PriceRule, error)
+	Update(PriceRule) (*PriceRule, error)
+	List() ([]PriceRule, error)
+	Delete(int64) error
+}
+
+// PriceRuleServiceOp handles communication with the price rule related methods of the Shopify API.
+type PriceRuleServiceOp struct {
+	client *Client
+}
+
+// PriceRule represents a Shopify discount rule
+type PriceRule struct {
+	ID                                     int64              `json:"id,omitempty"`
+	Title                                  string             `json:"title,omitempty"`
+	ValueType                              string             `json:"value_type,omitempty"`
+	Value                                  *decimal.Decimal   `json:"value,omitempty"`
+	CustomerSelection                      string             `json:"customer_selection,omitempty"`
+	TargetType                             string             `json:"target_type,omitempty"`
+	TargetSelection                        string             `json:"target_selection,omitempty"`
+	AllocationMethod                       string             `json:"allocation_method,omitempty"`
+	AllocationLimit                        string             `json:"allocation_limit,omitempty"`
+	OncePerCustomer                        bool               `json:"once_per_customer,omitempty"`
+	UsageLimit                             int32              `json:"usage_limit,omitempty"`
+	StartsAt                               *time.Time         `json:"starts_at,omitempty"`
+	EndsAt                                 *time.Time         `json:"ends_at,omitempty"`
+	CreatedAt                              *time.Time         `json:"created_at,omitempty"`
+	UpdatedAt                              *time.Time         `json:"updated_at,omitempty"`
+	EntitledProductIds                     []int64            `json:"entitled_product_ids,omitempty"`
+	EntitledVariantIds                     []int64            `json:"entitled_variant_ids,omitempty"`
+	EntitledCollectionIds                  []int64            `json:"entitled_collection_ids,omitempty"`
+	EntitledCountryIds                     []int64            `json:"entitled_country_ids,omitempty"`
+	PrerequisiteProductIds                 []int64            `json:"prerequisite_product_ids,omitempty"`
+	PrerequisiteVariantIds                 []int64            `json:"prerequisite_variant_ids,omitempty"`
+	PrerequisiteCollectionIds              []int64            `json:"prerequisite_collection_ids,omitempty"`
+	PrerequisiteSavedSearchIds             []int64            `json:"prerequisite_saved_search_ids,omitempty"`
+	PrerequisiteCustomerIds                []int64            `json:"prerequisite_customer_ids,omitempty"`
+	PrerequisiteSubtotalRange              *PrerequisiteRange `json:"prerequisite_subtotal_range,omitempty"`
+	PrerequisiteQuantityRange              *PrerequisiteRange `json:"prerequisite_quantity_range,omitempty"`
```

Comment:
The [docs](https://shopify.dev/docs/admin-api/rest/reference/discounts/pricerule?api[version]=2020-07) have `prerequisite_quantity_range.greater_than_or_equal_to` as a json number however we're parsing it into a property on a struct that is of type string, I believe this will blow up. I think the tests didn't catch this as we are using null values for a number of fields.

Reply:
I think you're right.
and 've fixed at f73da7fce26a9595434907899b3a9bf6210d5a7d .